Build failure in vcpkg-tool

**Describe the bug**
Unable to bootstrap vcpkg due to bug in vcpkg-tool

**Environment**
- OS: Ubuntu Bionic from docker repository
- Compiler: g++ 7.5.0
- vcpkg hash: 111220b3cf

**To Reproduce**
git clone https://github.com/microsoft/vcpkg.git
./bootstrap-vcpkg.sh -disableMetrics

**Expected behavior**
vcpkg built and installed,

**Failure logs** 
```
/opt/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-07-16/src/vcpkg/base/files.cpp:1365:28: error: expected primary-expression before ';' token
                 return path;
```

**Additional context**
See (vcpkg-tool)/src/vcpkg/base/files.cpp function "virtual path absolute(...)". The function does a "return path;" but path is a type. Looking at the file history I can see what went wrong. The old version had a return type fs::path and the first argument to the function was confusingly named path. Now the return type is "path" and the first argument has been renamed "target". So, the code should have done "return target;" I don't know whether that is the only issue. The problem does not apply to newer linux distros. I guess they end up with  with VCPKG_USE_STD_FILESYSTEM=1 which will skip the offending code.

+1

I see the same on:
```
Description:	Ubuntu 16.04.7 LTS
Release:	16.04
Codename:	xenial
```

Also with
```
g++ (Ubuntu 7.5.0-3ubuntu1~16.04) 7.5.0
```
that the vcpkg build properly finds along with `-lstdc++fs`:
```Building vcpkg-tool...
-- The CXX compiler identification is GNU 7.5.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++-7 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting the C++ compiler in use
-- Detecting the C++ compiler in use - gcc
-- Detecting the C++ standard library
-- Detecting the C++ standard library - libstdc++
-- Detecting how to use the C++ filesystem library
-- Detecting how to use the C++ filesystem library - <experimental/filesystem> with -lstdc++fs
...
```

A slightly fuller trace:
```
FAILED: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o 
/usr/bin/g++-7 -DVCPKG_BASE_VERSION=2021-07-16 -DVCPKG_USE_STD_FILESYSTEM=0 -DVCPKG_VERSION=unknownhash -I/home/circleci/project/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-07-16/include -O3 -DNDEBUG -include /home/circleci/project/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-07-16/include/pch.h -pthread -std=c++1z -MD -MT CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o -MF CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o.d -o CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o -c /home/circleci/project/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-07-16/src/vcpkg/base/files.cpp
/home/circleci/project/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-07-16/src/vcpkg/base/files.cpp: In member function ‘virtual std::experimental::filesystem::v1::__cxx11::path vcpkg::RealFilesystem::absolute(const std::experimental::filesystem::v1::__cxx11::path&, std::error_code&) const’:
/home/circleci/project/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-07-16/src/vcpkg/base/files.cpp:1365:28: error: expected primary-expression before ‘;’ token
                 return path;
                            ^
```

--- a/scripts/bootstrap.ps1
+++ b/scripts/bootstrap.ps1
@@ -45,7 +45,7 @@ while (!($vcpkgRootDir -eq "") -and !(Test-Path "$vcpkgRootDir\.vcpkg-root"))
 
 Write-Verbose "Examining $vcpkgRootDir for .vcpkg-root - Found"
 
-& "$scriptsDir/tls12-download.exe" github.com "/microsoft/vcpkg-tool/releases/download/2021-07-16/vcpkg.exe" "$vcpkgRootDir\vcpkg.exe"
+& "$scriptsDir/tls12-download.exe" github.com "/microsoft/vcpkg-tool/releases/download/2021-07-21/vcpkg.exe" "$vcpkgRootDir\vcpkg.exe"
 Write-Host ""
 
 if ($LASTEXITCODE -ne 0)

--- a/scripts/bootstrap.sh
+++ b/scripts/bootstrap.sh
@@ -278,8 +278,8 @@ else
 fi
 
 # Do the build
-vcpkgToolReleaseTag="2021-07-16"
-vcpkgToolReleaseSha="e86a70ea63c79fd86099af271e7aff7d1436e703a2bcb56b5671d1ff49924d44e2fa0d8ab64684b20d50d0872d7b4eb2c7b0800899eedf19f51bf11bf3a4fb44"
+vcpkgToolReleaseTag="2021-07-21"
+vcpkgToolReleaseSha="9d1644c1c34bfd750cad36c47d7997698b0c5c4ff0673f9f39ff5f62f6e351ca32db7991043e7f154aa8ffe8b5180591d78131edf5dc533164d75a60764a8da8"
 vcpkgToolReleaseTarball="$vcpkgToolReleaseTag.tar.gz"
 vcpkgToolUrl="https://github.com/microsoft/vcpkg-tool/archive/$vcpkgToolReleaseTarball"
 baseBuildDir="$vcpkgRootDir/buildtrees/_vcpkg"

